source/uwp/Renderer/lib/XamlCardRendererComponent.cpp

```diff
@@ -231,4 +265,45 @@ namespace AdaptiveCards { namespace XamlCardRenderer
         return m_explicitDimensions;
     }
 
+    _Use_decl_annotations_
+    HRESULT XamlCardRenderer::get_ElementRenderers(IAdaptiveElementRendererRegistration **value)
+    {
+        *value = m_elementRendererRegistration.Get();
+        return S_OK;
+    }
+    _Use_decl_annotations_
+    HRESULT XamlCardRenderer::get_ActionRenderers(IAdaptiveActionRendererRegistration **value)
+    {
+        *value = m_actionRendererRegistration.Get();
+        return S_OK;
+    }
+
+    _Use_decl_annotations_
+    HRESULT XamlCardRenderer::RegisterDefaultElementRenderers()
+    {
+        RETURN_IF_FAILED(m_elementRendererRegistration->Set(HStringReference(L"Input.ChoiceSet").Get(), ComPtr<IAdaptiveElementRenderer>(Make<AdaptiveChoiceSetInputRenderer>().Get()).Get()));
```

Comment:
> RETURN_IF_FAILED(m_ [](start = 7, length = 20)

Any ideas on another way to do this easily? I'm not a fan of how this ended up looking.

Reply:
I think you've got one more layer of ComPtr than you need.  This seems to work:
RETURN_IF_FAILED(m_elementRendererRegistration->Set(HStringReference(L"Input.ChoiceSet").Get(), (Make<AdaptiveChoiceSetInputRenderer>().Get())));

---
In reply to: [141197285](https://github.com/Microsoft/AdaptiveCards/pull/695#discussion_r141197285) [](ancestors = 141197285)

Reply:
I seem to recall it was complaining about Make Returning a ComPtr of the class rather than the interface. I'll give that another try. thanks for checking!

---
In reply to: [141431783](https://github.com/Microsoft/AdaptiveCards/pull/695#discussion_r141431783) [](ancestors = 141431783,141197285)

Reply:
Yay!. That worked just fine. Not sure what I was doing differently yesterday that made me go this route.

---
In reply to: [141432272](https://github.com/Microsoft/AdaptiveCards/pull/695#discussion_r141432272) [](ancestors = 141432272,141431783,141197285)